Tasks/Common/nuget-task-common/BuildMetadataHelpers.ts

```diff
@@ -0,0 +1,87 @@
+import Q = require('q');
+import tl = require('vsts-task-lib/task');
+import vstsWebApi = require('vso-node-api/WebApi');
+
+const fs = require('fs');
+const JSZip  = require('jszip');
+const xmlreader = require('xmlreader');
+const httpm = require('vso-node-api/HttpClient');
+const restm = require('vso-node-api/RestClient');
+
+export interface PackageBuildMetadata {
+    PackageName: string;
+    ProtocolType: string;
+    BuildId: string;
+    CommitId: string;
+    BuildCollectionId: string;
+    BuildProjectId: string;
+    RepositoryId: string;
+    BuildAccountId: string; //NOTE: NULL for on prem, must set this value for hosted!
+    OriginalPackageVersion: string;
+}
+
+export interface PackageMetadata {
+    Name: string;
+    Version: string;
+}
+
+function getNuspec(packageFile: string): Q.Promise<string> {
+    var defer = Q.defer<string>();    
+    fs.readFile(packageFile, (err, data) => { //read packageFile
+        JSZip.loadAsync(data) //unzip nupkg
+        .then(nupkg => {
+            return nupkg.file(/\.nuspec/)[0].async('string') //read nuspec (NOTE: reads the first nuspec found in a nupkg)
+            .then(nuspec => {
+                nuspec = nuspec.trim(); //trim whitespace
+                defer.resolve(nuspec);
+            });
+        });
+    });
+    return defer.promise;
+}
+
+export function getPackageMetadata(packageFile: string): Q.Promise<PackageMetadata> {
+    var defer = Q.defer<PackageMetadata>(); 
+    getNuspec(packageFile)
+    .then(nuspec => {
+        xmlreader.read(nuspec, (err, res) => {
+            defer.resolve({
+                'Name': res.package.metadata.id.text(),
+                'Version': res.package.metadata.version.text()
+            });
+        });
+    });
+    return defer.promise;
+}
+
+export function getFeedName(feedUrl: string): string {
+    var url = feedUrl.split('/');
+    var nextValueIsFeedName = false;
+    for (var value of url) {
+        if (nextValueIsFeedName) {
+            return value;
+        }
+        if (value == '_packaging') {
+            nextValueIsFeedName = true;
+        } 
+    }
+}
```

Comment:
I would explicitly return null in the error case here (rather than implicitly returning undefined by not returning anything)